### PR TITLE
Use github actions in replacement of CircleCI

### PR DESCRIPTION
**Pull request by cclerget**
Motivations in favor of github actions:

- Integrated CI for contributors
- Github actions are not subject to Docker Hub rate limit which is convenient for e2e tests
- More performant than CircleCI free plan

### CODE REVIEW

**Review by ikaneshiro — APPROVED** [review]
Lgtm

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -0,0 +1,204 @@
...
+      - name: Check code
+        run: make -C ./builddir check
+  
+  rpmbuild-centos7:
+    runs-on: ubuntu-20.04
+    name: rpmbuild-centos7
+    steps:
+      - uses: actions/checkout@v2
+
+      - name: Set OS type and version
```
Comment (DrDaveD) [inline .github/workflows/ci.yml]: This step does more than set OS type and version, it actually runs the build.  That was confusing when I went to look at the output.

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -0,0 +1,204 @@
...
+          OS_VERSION: 7
+        run: ./scripts/ci-docker-run
+
+  rpmbuild-centos8:
+    runs-on: ubuntu-20.04
+    name: rpmbuild-centos8
+    steps:
+      - uses: actions/checkout@v2
+
+      - name: Set OS type and version
```
Comment (DrDaveD) [inline .github/workflows/ci.yml]: Same here

**Review by DrDaveD — CHANGES_REQUESTED** [review]
Please change the name of the rpmbuild build step, otherwise looks good.

File: .github/workflows/ci.yml (hunk 1 above)
Comment (cclerget) [reply .github/workflows/ci.yml]: fixed

**Review by DrDaveD — APPROVED** [review]

**Pull request merged**